src/OrchardCore.Modules/OrchardCore.Contents/Views/Contents-Button-Actions.SummaryAdmin.cshtml

```diff
@@ -6,32 +6,17 @@
     var hasPublished = await ContentManager.HasPublishedVersionAsync(contentItem);
     var hasDraft = contentItem.HasDraft();
     var hasClonePermission = await AuthorizationService.AuthorizeAsync(User, OrchardCore.Contents.Permissions.CloneContent, contentItem);
+    var hasDeletePermission = await AuthorizationService.AuthorizeAsync(User, OrchardCore.Contents.Permissions.DeleteContent, contentItem);
+    var hasEditPermission = await AuthorizationService.AuthorizeAsync(User, OrchardCore.Contents.Permissions.EditContent, contentItem);
+    var hasPreviewPermission = await AuthorizationService.AuthorizeAsync(User, OrchardCore.Contents.Permissions.PreviewContent, contentItem);
+    var hasPublishPermission = await AuthorizationService.AuthorizeAsync(User, OrchardCore.Contents.Permissions.PublishContent, contentItem);
 }
 
-@if (hasDraft)
-{
-    <a asp-action="Publish" asp-controller="Admin" asp-route-area="OrchardCore.Contents" asp-route-contentItemId="@contentItem.ContentItemId" asp-route-returnUrl="@FullRequestPath" itemprop="UnsafeUrl" class="dropdown-item btn-sm">@T["Publish Draft"]</a>
-    <a asp-action="Preview" asp-controller="Item" asp-route-area="OrchardCore.Contents" asp-route-contentItemId="@contentItem.ContentItemId" class="dropdown-item btn-sm">@T["Preview Draft"]</a>
+<a asp-action="Publish" asp-controller="Admin" asp-route-area="OrchardCore.Contents" asp-route-contentItemId="@contentItem.ContentItemId" asp-route-returnUrl="@FullRequestPath" itemprop="UnsafeUrl" class="dropdown-item btn-sm @(hasDraft && hasPublishPermission)? string.Empty : "disabled")">@T["Publish Draft"]</a>
+<a asp-action="Preview" asp-controller="Item" asp-route-area="OrchardCore.Contents" asp-route-contentItemId="@contentItem.ContentItemId" class="dropdown-item btn-sm @(hasDraft && hasPreviewPermission? string.Empty : "disabled")">@T["Preview Draft"]</a>
+<div class="dropdown-divider"></div>
 
-    <div class="dropdown-divider"></div>
-}
-
-@if (hasDraft && hasPublished)
-{
-    <a asp-action="DiscardDraft" asp-controller="Admin" asp-route-area="OrchardCore.Contents" asp-route-contentItemId="@contentItem.ContentItemId" asp-route-returnUrl="@FullRequestPath" itemprop="UnsafeUrl RemoveUrl" data-title=@T["Discard Draft"] data-message=@T["Are you sure you want to discard this draft?"] data-ok-text="@T["Yes"]" data-cancel-text="@T["No"]" data-ok-class="btn-primary" class="dropdown-item btn-sm">@T["Discard Draft"]</a>
-}
-
-@if (hasPublished)
```

Comment:
thank you @hishamco the pr becomes more understandable now.

So you are changing all of these to display as disabled, rather than not displaying at all.

I don't think that's the way to go here (but if it is there are many more permission checks you would have to change)

I would rather see it as it was, the button not shown.

i.e. here you are showing discard draft, where there is no draft to discard, which makes no sense.

Reply:
> So you are changing all of these to display as disabled, rather than not displaying at all.

I was think about that before, but hide them will make the buttons look ugly, so I prefer to disable them rather than hide them

> I would rather see it as it was, the button not shown.

Let me check them again, they was working well before I remove the check that you asked for, so let me check everything again

Thanks Dean